### PR TITLE
(Feature) add AdvancedMarker component

### PR DESCRIPTION
**Pull request by VictorBalbo**
Added `AdvancedMarker` component as it is the replacement for `Marker` component.
Marker component still exists, the new component was added just as a new option

### CODE REVIEW

File: docs/components/avanced-marker.md
Comment (MichaelVini) [inline docs/components/avanced-marker.md:1]: Shouldn't the file name be advanced-marker.md?

**Comment by waledagne** [comment]
@MichaelVini Will this be merged anytime soon?

File: docs/components/avanced-marker.md
Comment (VictorBalbo) [reply docs/components/avanced-marker.md:1]: Yes, it was a typo

**Comment by MichaelVini** [comment]
> @MichaelVini Will this be merged anytime soon?

I don't have permission to merge the PR I just added a comment with a question. However, I would like this to be merged as I also need it 😄

File: docs/.vitepress/config.mts [hunk 1]
```diff
@@ -19,6 +19,10 @@ export default defineConfig({
         link: "/components/",
         collapsed: false,
         items: [
+          {
+            text: "Advanced Marker",
+            link: "/components/avanced-marker",
```
Comment (MichaelVini) [inline docs/.vitepress/config.mts:24]:
```suggestion
            link: "/components/advanced-marker",
```

File: docs/components/advanced-marker.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+<script setup>
+import { GoogleMap, AdvancedMarker } from '@lib'
+import { apiPromise } from '@docs/shared'
+
+const center = { lat: 40.689247, lng: -74.044502 }
+</script>
+
+# AvancedMarker
```
Comment (MichaelVini) [inline docs/components/advanced-marker.md:8]:
```suggestion
# AdvancedMarker
```

File: docs/components/advanced-marker.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+Use the `AvancedMarker` component to draw markers, drop pins or any custom icons on a map. `AvancedMarker` is the new version offered by google when deprecated the `Marker` component ([read more here](https://developers.google.com/maps/deprecations#googlemapsmarker_in_the_deprecated_as_of_february_2024)).
```
Comment (MichaelVini) [inline docs/components/advanced-marker.md:10]:
```suggestion
Use the `AdvancedMarker` component to draw markers, drop pins or any custom icons on a map. `AvancedMarker` is the new version offered by google when deprecated the `Marker` component ([read more here](https://developers.google.com/maps/deprecations#googlemapsmarker_in_the_deprecated_as_of_february_2024)).
```

File: docs/components/index.md [hunk 4]
```diff
@@ -4,6 +4,7 @@ This library is intended to be used in a composable fashion. Therefore you will
 
 The main mapping component is `GoogleMap`, however the following components are available at your disposal:
 
+- [AvancedMarker](./avanced-marker.md)
```
Comment (MichaelVini) [inline docs/components/index.md:7]:
```suggestion
- [AdvancedMarker](./advanced-marker.md)
```

**Review by HusamElbashir — APPROVED** [review]
Thanks for the contribution @VictorBalbo !

**Pull request merged**